feat: Implement .albumprops inheritance system

adamkovacs2000 opened the pull request:
Add comprehensive inheritance functionality for .albumprops files:

Features:
- Chainable property inheritance from parent to child directories
- inherit=True/False to control inheritance flow
- inherit_properties to specify which properties to inherit
- Special share_with logic for user permission inheritance
- Final album properties logging for debugging
- Full backward compatibility

Technical Details:
- New AlbumModel attributes: inherit, inherit_properties
- build_inheritance_chain_for_album_path() for directory traversal
- apply_inheritance_to_album_model() for property merging
- merge_inherited_share_with() for complex user handling
- Updated build_album_list() to use inheritance for all albums

Salvoxia commented:
Thanks for your response.
I played around with it a bit and tested it with one of my usual "ugly" testing scenarios, looking like this:
`/photos/2023/Christmas/.albumprops`:
```yaml
  override_name: "Family Photos"
  description: "Family photos"
  inherit: true
  share_with:
    - user: "dad"
      role: "editor"
```

`/photos/2023/Christmas/Cookies/.albumprops`:
```yaml
  inherit: true
  share_with:
    - user: "mom"
      role: "viewer"
```

`/photos/2023/Vacation/.albumprops`:
```yaml
  override_name: "Family Photos"
  description: "Family photos"
  share_with:
    - user: "dad"
      role: "editor"
```

This is a very nasty case where photos from different folders on the same level are merged into the same album with `override_name`, but now with inheritance.

The resulting album `Family Photos` was __not__ shared with `mom`.
Is that what you would expect?

`override_name` creates all sorts of funny situations. I guess implementing property precedence while respecting the folder level across different folder trees just because they are merged to the same album with the same `override_name` gets ugly.

Salvoxia commented on README.md:
````diff
@@ -711,9 +716,108 @@ sort_oder: "desc"
+
+**Example 1: Basic Inheritance**
+```
+/photos/.albumprops:
````
I think it would improve readability if you would put the file path into an inline code-block and split different `.albumprops` file examples into individual code blocks with YAML syntax highlighting like this:
`/photos/.albumprops:`
```yaml
inherit: true
  description: "Family photos"
  share_with:
    - user: "dad"
      role: "editor"
```

`/photos/2023/.albumprops:`
```yaml
inherit: true
  share_with:
    - user: "mom"
      role: "viewer"
```
Result for `/photos/2023/vacation/`:
```yaml
# - description: "Family photos" (inherited)
# - share_with: dad (editor), mom (viewer)
```

adamkovacs2000 replied:
Done!

adamkovacs2000 commented:
Hi! 
I updated the readme and even extended it with the latest additions.
Also fixed the linting errors as far as i know (the github process hasn't run yet).
